src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -1100,9 +1100,7 @@ public async ValueTask<HttpResponseMessage> SendWithVersionDetectionAndRetryAsyn
                                     await connection.InitialSettingsReceived.WaitWithCancellationAsync(cancellationToken).ConfigureAwait(false);
                                     if (!connection.IsConnectEnabled)
                                     {
-                                        HttpRequestException exception = new(SR.net_unsupported_extended_connect);
-                                        exception.Data["SETTINGS_ENABLE_CONNECT_PROTOCOL"] = false;
```

Comment:
This should stay (https://github.com/dotnet/runtime/pull/88974#discussion_r1264933744).